docs/providers/aws/guide/functions.md

```diff
@@ -216,13 +216,17 @@ provider:
       baseimage:
         path: ./path/to/context
         file: Dockerfile.dev
+        buildArgs:
+          STAGE: ${opt:stage}
       anotherimage:
         uri: 000000000000.dkr.ecr.sa-east-1.amazonaws.com/test-lambda-docker@sha256:6bb600b4d6e1d7cf521097177dd0c4e9ea373edb91984a505333be8ac9455d38
 ```
 
 When configuring functions, images should be referenced via `image` property, which can point to an image already defined in `provider.ecr.images` or directly to an existing AWS ECR image, following the same format as `uri` above.
 Both `handler` and `runtime` properties are not supported when `image` is used.
 
+Dockerfile `docker build` arguments (`--build-arg`) may be passed in so that your Dockerfile can use and reference `ARG` values withing your `Dockerfile`. These are passed in using the `buildArg` configuration section. This cannot be paired with `uri` configurations.
```

Comment:
Let's add these information as a continuation of description from line 206. In its current form it's quite confusing, as it's placed between a description of `image` property on a function and an example of using `image` property that isn't in any way related to definition of `buildArgs`. 

Maybe let's go with something like that in a new line after line 206.

```
Additionally, with `buildArgs` property, you can define arguments that will be passed to `docker build` command with `--build-arg` flag. They might be later referenced via `ARG` within your `Dockerfile`. When `uri` is defined for an image, `buildArgs` cannot be defined.
```

Reply:
love it - I'll update the branch.